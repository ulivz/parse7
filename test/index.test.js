import parse7 from '../src'

const RAW = {
  simple: `<div class="content">Chars</div>`,
  singleClosed: '<app class="content"/><div id="1">Chars</div>',
  comments: `<div class="ddd">Chars</div><!-- I am Commnets -->`,
  nested: `<a class="content">Chars<b>CCC</b></a>`,
}

function getResult(s) {
  const result = []
  parse7(s, {
    tagStart(tagName, attrs) {
      result.push(tagName)
      result.push(attrs)
    },
    tagEnd(tagName) {
      result.push(tagName)
    },
    comments(comment) {
      result.push(comment)
    },
    chars(chars) {
      result.push(chars)
    }
  })
  return result
}

test('normal', () => {
  const r = getResult(RAW.simple)
  expect(r).toMatchSnapshot()
})

test('singleClosed', () => {
  const r = getResult(RAW.singleClosed)
  expect(r).toMatchSnapshot()
})

test('comments', () => {
  const r = getResult(RAW.comments)
  expect(r).toMatchSnapshot()
})

test('nested', () => {
  const r = getResult(RAW.nested)
  expect(r).toMatchSnapshot()
})
