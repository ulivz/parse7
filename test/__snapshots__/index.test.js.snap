// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments 1`] = `
Array [
  "div",
  Array [
    Object {
      "name": "class",
      "value": "ddd",
    },
  ],
  "Chars",
  "div",
  " I am Commnets ",
]
`;

exports[`nested 1`] = `
Array [
  "a",
  Array [
    Object {
      "name": "class",
      "value": "content",
    },
  ],
  "Chars",
  "b",
  Array [],
  "CCC",
  "b",
  "a",
]
`;

exports[`normal 1`] = `
Array [
  "div",
  Array [
    Object {
      "name": "class",
      "value": "content",
    },
  ],
  "Chars",
  "div",
]
`;

exports[`singleClosed 1`] = `
Array [
  "app",
  Array [
    Object {
      "name": "class",
      "value": "content",
    },
  ],
  "app",
  "div",
  Array [
    Object {
      "name": "id",
      "value": "1",
    },
  ],
  "Chars",
  "div",
]
`;
